packages/salesforcedx-vscode-soql/src/editor/queryRunner.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { JsonMap } from '@salesforce/ts-types';
+import { QueryResult } from 'jsforce';
+import * as vscode from 'vscode';
+
+export class QueryRunner {
+  constructor(
+    private connection: Connection,
+    private document?: vscode.TextDocument
+  ) {}
+
+  public async runQuery(queryText: string): Promise<JsonMap[]> {
+    try {
+      const rawQueryData = (await this.connection.query(
```

Comment:
should we have a check before even attempting to run the query to see if there are any known syntax errors? I think that check has to be in the caller.

Reply:
I'm not sure how we would check for syntax error based on the string text from the document. Wouldn't that require a parser and models that now live the UI?

If the query has bad syntax or can't be run then at least the user is notified thats the case and directed to fix it by the error message.

Reply:
I think in the current design, you may have to call the SOQL parser directly to validate that. With that being said I think we shouldn't have to worry about it now.
It is something that we could think about later once we have the language server for SOQL working. The language server should be able to tell us at the minimum whether a given soql text has any syntax errors are not. It is an optimization we could consider in the future and consider disabling the RUN Query button in those scenarios with errors.